pkg/webhooks/generation.go

```diff
@@ -193,8 +193,8 @@ func stripNonPolicyFields(obj, newRes map[string]interface{}, logger logr.Logger
 
 	if _, found := obj["metadata"]; found {
 		requiredMetadataInObj := make(map[string]interface{})
-		if _, found := obj["metadata"].(map[string]interface{})["annotations"]; found {
-			requiredMetadataInObj["annotations"] = obj["metadata"].(map[string]interface{})["annotations"]
+		if annotations, found := obj["metadata"].(map[string]interface{})["annotations"]; found {
+			requiredMetadataInObj["annotations"] = annotations
```

Comment:
Same improvement can be made here: https://github.com/kyverno/kyverno/blob/c7b50bf0b1eb9271cb06d37ec10bc00fe0cbad94/pkg/webhooks/generation.go#L200

Please change all other places that use this pattern.